lib/dynamoid/transaction/create.rb

```diff
@@ -0,0 +1,37 @@
+require_relative 'action'
+
+module Dynamoid
+  module Transaction
+    class Create < Action
+
+      def run_callbacks
+        (yield; return) unless model
+        model.run_callbacks(:save) do
+          model.run_callbacks(:create) do
+            model.run_callbacks(:validate) do
+              yield
+            end
+          end
+        end
+      end
+
+      def to_h
+        item = attributes.clone
+        item = add_timestamps(item)
+        item = model.attributes.merge(item) if model
+        item[model_class.hash_key] ||= SecureRandom.uuid
+        item = Dynamoid::Dumping.dump_attributes(item, model_class.attributes)
```

Comment:
This code assumes Symbol keys in `attributes`. Dynamoid accepts String keys as well usually.

Reply:
Field values should also be type casted, not only dumped so use could write `create(age: "12")`.